File: Anguler/SCM_Anguler/src/app/component/shared/model/warehouse.ts
export interface WarehouseRequestModel {
  name: string;
  email: string;
  location: string;
  address:string;
  capacity: number;
  managerId: number;
  isActive: boolean;
  policeStationId: number;
}

export interface WarehouseResponseModel {
  id: number;
  name: string;
  email: string;
  location: string;
  address:string;
  capacity: number;
  managerId: number;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  policeStationId: number;
  policeStationName: string;
  districtName: string;
  divisionName: string;
}

File: sp-2 security/SCM/src/main/java/com/example/SCM/entity/Warehouse.java
package com.example.SCM.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "warehouses")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Warehouse {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 150)
    private String name;

    @Column(nullable = false, unique = true)
    private String email;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String location;

    private String address;

    private double capacity;

    private Long managerId;

    private boolean isActive = true;

    @Column(updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "police_station_id")
    private PoliceStation policeStation;

    @JsonIgnore
    @ManyToMany(mappedBy = "warehouses", fetch = FetchType.LAZY)
    private Set<Driver> drivers = new HashSet<>();
}